src/messages/manifestjson.js

```diff
@@ -106,6 +106,20 @@ export function manifestPropMissing(property) {
   };
 }
 
+export const MANIFEST_ICON_NOT_FOUND = 'MANIFEST_ICON_NOT_FOUND';
+export function manifestIconMissing(path) {
+  return {
+    code: MANIFEST_ICON_NOT_FOUND,
+    legacyCode: null,
+    message: _(
+      'An icon defined in the manifest could not be found in the package.'),
+    description: sprintf(
+      _('Icon could not be found at "%(path)s".'),
+      {path}),
```

Comment:
It looks like the unsupported API code is using `context.report()` which can use eslint's built-in string formatting. I think that works fine.

I was hoping to use it here but since the manifest validation updates the collector directly I don't think we have support for it.